File: code/community/1306409966936002566/debugging-403-error-gcp-bucket-signed-url.go
package main

import (
    "fmt"
    "io/ioutil"
    "log"
    "os"

    dg "github.com/deepgram-devs/deepgram-go-sdk/deepgram"
)

func main() {
    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    filepath := "./audio-file.wav" // Path to your audio file

    // Initialize Deepgram client
    client := dg.NewClient(apiKey)

    // Read audio file
    audioData, err := ioutil.ReadFile(filepath)
    if err != nil {
        log.Fatal(err)
    }

    // Transcribe audio
    options := map[string]interface{}{
        "language": "en-US",
    }
    response, err := client.TranscribeAudio(audioData, "audio/wav", options)
    if err != nil {
        log.Fatal(err)
    }

    // Print transcription
    transcript := response.Results.Channels[0].Alternatives[0].Transcript
    fmt.Println("Transcription:", transcript)
}

File: code/community/1306409966936002566/debugging-403-error-gcp-bucket-signed-url.py
import os
import requests
from deepgram import Deepgram

# Initialize Deepgram SDK
DEEPGRAM_API_KEY = os.environ['DEEPGRAM_API_KEY']
filepath = './audio-file.wav' # Path to your audio file

dg_client = Deepgram(DEEPGRAM_API_KEY)

# Read in your audio file data
with open(filepath, 'rb') as audio:
    audio_data = audio.read()

# Transcribe audio from file
transcription = dg_client.transcription.sync_prerecorded(
    buffer=audio_data,
    mimetype='audio/wav',
    options={'language': 'en-US'}
)

# Print the transcription
print(transcription['results']['channels'][0]['alternatives'][0]['transcript'])